pkg/backend/httpstate/backend.go

```diff
@@ -1080,13 +1090,21 @@ func (b *cloudBackend) apply(
 	}
 
 	// Create an update object to persist results.
-	update, version, token, err := b.createAndStartUpdate(ctx, kind, stack, &op, opts.DryRun)
+	update, updateMeta, err := b.createAndStartUpdate(ctx, kind, stack, &op, opts.DryRun)
 	if err != nil {
 		return nil, nil, result.FromError(err)
 	}
 
-	permalink := b.getPermalink(update, version, opts.DryRun)
-	return b.runEngineAction(ctx, kind, stack.Ref(), op, update, token, permalink, events, opts.DryRun)
+	// Display messages from the backend if present.
+	if len(updateMeta.messages) > 0 {
+		for _, msg := range updateMeta.messages {
+			fmt.Println(msg.Message)
+		}
+		fmt.Print("\n")
+	}
+
+	permalink := b.getPermalink(update, updateMeta.version, opts.DryRun)
+	return b.runEngineAction(ctx, kind, stack.Ref(), op, update, updateMeta.leaseToken, permalink, events, opts.DryRun)
```

Comment:
Not sure if there are tests to be added for this.

If you have any ideas of useful or good tests to add on this, I'd greatly appreciate the advice.